src/pages/libs/db.ts

```diff
@@ -1,65 +1,134 @@
-import Dexie, { Table } from "dexie";
+import { fetchData } from './storage'
+import { version } from '../../../package.json'
+import Browser from "webextension-polyfill";
+import cute from "@assets/img/memes/cute-128.png";
+
+import {
+  PROTOCOLS_API,
+  METAMASK_LIST_CONFIG_API,
+  DEFILLAMA_DIRECTORY_API,
+  PROTOCOL_TVL_THRESHOLD,
+} from "./constants";
 
 export interface Protocol {
-  name: string;
   url: string;
-  logo: string;
-  category: string;
   tvl?: number;
 }
 
-export class ProtocolsDb extends Dexie {
-  protocols!: Table<Protocol>;
-
-  constructor() {
-    super("ProtocolsDb");
-    this.version(1).stores({
-      protocols: "name, category",
-    });
-  }
+export const blockedDomainsDb: {
+  data: Set<string>
+} = {
+  data: new Set()
 }
 
-export const protocolsDb = new ProtocolsDb();
+export const fuzzyDomainsDb: {
+  data: string[]
+} = {
+  data: []
+}
 
-export interface Domain {
-  domain: string;
+export const allowedDomainsDb: {
+  data: Set<string>
+} = {
+  data: new Set()
 }
 
-export class AllowedDomainsDb extends Dexie {
-  domains!: Table<Domain>;
+const cacheKey = 'cache-v' + version
 
-  constructor() {
-    super("AllowedDomainsDb");
-    this.version(1).stores({
-      domains: "domain",
-    });
-  }
-}
+async function getData() {
+  console.time(cacheKey)
+  const rawProtocols = await fetch(PROTOCOLS_API).then((res) => res.json());
+  const protocols = (
+    (rawProtocols["protocols"]?.map((x: any) => ({
+      url: x.url,
+      tvl: x.tvl || 0,
+    })) ?? []) as Protocol[]
+  ).filter((x) => x.tvl >= PROTOCOL_TVL_THRESHOLD);
```

Comment:
this is for later but we should do something to include protocols with no tvl but that are significant, for example the metamask or 1inch listings